crates/modelardb_server/tests/integration_test.rs

```diff
@@ -76,45 +75,42 @@ enum TableType {
     ModelTableAsField,
 }
 
-/// Async runtime, handler to the server process, and client for use by the tests. It implements
-/// `drop()` so the resources it manages is released no matter if a test succeeds, fails, or panics.
+/// Handler to the server process and client for use by the tests. The local folder is deleted and
+/// the server process is killed when the struct is dropped.
 struct TestContext {
     temp_dir: TempDir,
     port: u16,
-    runtime: Runtime,
     server: Child,
     client: FlightServiceClient<Channel>,
 }
 
 impl TestContext {
-    /// Create a [`Runtime`], a server that stores data in a randomly generated local data folder
-    /// and listens on `PORT`, and a client and ensure they are all ready to be used in the tests.
-    fn new() -> Self {
-        let runtime = Runtime::new().unwrap();
+    /// Create a server that stores data in a randomly generated local data folder and listens on
+    /// `PORT` and a client that is connected to the server.
+    async fn new() -> Self {
         let temp_dir = tempfile::tempdir().unwrap();
         let port = PORT.fetch_add(1, Ordering::Relaxed);
-        let mut server = Self::create_server(&temp_dir, port);
-        let client = Self::create_client(&runtime, &mut server, port);
+        let server = Self::create_server(&temp_dir, port).await;
+        let client = Self::create_client(port).await;
 
         Self {
             temp_dir,
             port,
-            runtime,
             server,
             client,
         }
     }
 
-    /// Restart the server and reconnect the client.
-    fn restart_server(&mut self) {
-        Self::kill_child(&mut self.server);
-        self.server = Self::create_server(&self.temp_dir, self.port);
-        self.client = Self::create_client(&self.runtime, &mut self.server, self.port);
+    /// Restart the server on a new port and reconnect the client.
```

Comment:
Maybe add an additional sentence stating why a new port is used.

Reply:
Fixed.